src/macaron/slsa_analyzer/checks/check_result.py

```diff
@@ -24,25 +29,54 @@ class CheckResultType(str, Enum):
     UNKNOWN = "UNKNOWN"
 
 
-class CheckResult(TypedDict):
-    """This class stores the result of a check in a dictionary."""
+@dataclass
+class CheckInfo:
+    """This class identifies and describes a check."""
 
     check_id: str
     check_description: str
-    # The string representations of the slsa requirements and their
-    # corresponding slsa level.
-    slsa_requirements: list[str]
+    eval_reqs: list[ReqName]
+
+
+@dataclass
+class CheckResultData:
```

Comment:
Adding docstrings for these classes may be a good idea.
You can follow [instructions here](https://oracle.github.io/macaron/pages/developers_guide/style_guide.html#docstrings-for-classes) to write docstrings that are compatible with our Sphinx setup (the second bullet point is for `dataclass` classes).

Reply:
I've added doc comments for the fields.